pkgs/desktops/gnome/extensions/guillotine/default.nix

```diff
@@ -0,0 +1,46 @@
+{
+  stdenv,
+  lib,
+  fetchFromGitLab
+}:
+
+stdenv.mkDerivation rec {
+  pname = "gnome-shell-extension-guillotine";
+  version = "24";
+
+  src = fetchFromGitLab {
+    owner = "ente76";
+    repo = "guillotine";
+    rev = "v${version}";
+    hash = "sha256-eNhK3h9luPGXHR3lPkfu/mUN9+ixma64rbCk0cjF4Fc=";
+  };
+
+  dontBuild = true;
+
+  passthru = {
+    extensionUuid = "guillotine@fopdoodle.net";
+    extensionPortalSlug = "guillotine";
+  };
+
+  installPhase = ''
+    runHook preInstall
+    mkdir -p $out/share/gnome-shell/extensions/guillotine@fopdoodle.net
+    cp -R schemas "$out/share/gnome-shell/extensions/guillotine@fopdoodle.net"
+    cp default.json $out/share/gnome-shell/extensions/guillotine@fopdoodle.net
+    cp extension.js "$out/share/gnome-shell/extensions/guillotine@fopdoodle.net"
+    cp guillotine-symbolic.svg "$out/share/gnome-shell/extensions/guillotine@fopdoodle.net"
+    cp LICENSE "$out/share/gnome-shell/extensions/guillotine@fopdoodle.net"
+    cp metadata.json "$out/share/gnome-shell/extensions/guillotine@fopdoodle.net"
+    cp README.md "$out/share/gnome-shell/extensions/guillotine@fopdoodle.net"
+    runHook postInstall
+  '';
+
+  meta = with lib; {
+    description = "Guillotine is a gnome extension designed for efficiently carrying out executions of commands from a customizable menu. Simply speaking: it is a highly customizable menu that enables you to launch commands and toggle services.";
+    homepage = "https://gitlab.com/ente76/guillotine/";
+    license = licenses.gpl3;
```

Comment:
`gpl3` is unclear, use `gpl3Plus` or `gpl3Only`. See https://discourse.nixos.org/t/lib-licenses-gpl3-co-are-now-deprecated/8206 for more details.